src/cpp/fastdds/xtypes/dynamic_types/idl_parser/IdlParser.hpp

```diff
@@ -0,0 +1,2210 @@
+        }
+        else
+        {
+            // Keep the identifier for super-expression use
+            state["identifier"] = identifier_name;
+        }
+    }
+
+};
+
+template<>
+struct action<scoped_name>
+{
+    template<typename Input>
+    static void apply(
+            const Input& in,
+            Context* ctx,
+            std::map<std::string, std::string>& state,
+            std::vector<traits<DynamicData>::ref_type>& operands)
+    {
+        Module& module = ctx->module();
+        std::string identifier_name = in.string();
+
+        if (state.count("arithmetic_expr"))
+        {
+            if (module.has_constant(identifier_name))
+            {
+                DynamicData::_ref_type xdata = module.constant(identifier_name);
+                operands.push_back(xdata);
+            }
+            else
+            {
+                EPROSIMA_LOG_ERROR(IDLPARSER, "Unknown constant or identifier: " << identifier_name);
+                throw std::runtime_error("Unknown constant or identifier: " + identifier_name);
+            }
+        }
+    }
+
+};
+
+template<>
+struct action<semicolon>
+{
+    template<typename Input>
+    static void apply(
+            const Input& /*in*/,
+            Context* /*ctx*/,
+            std::map<std::string, std::string>& state,
+            std::vector<traits<DynamicData>::ref_type>& /*operands*/)
+    {
+        if (!state["type"].empty() && state.count("current_struct_member_name") &&
+                !state["current_struct_member_name"].empty())
+        {
+            // Add the type and name to the member lists
+            state["struct_member_types"] += state["type"] + ";";
+            state["struct_member_names"] += state["current_struct_member_name"] + ";";
+
+            // Add the array dimensions for this member to `all_array_sizes`
+            std::string current_array_sizes = state["current_array_sizes"].empty() ? "0" : state["current_array_sizes"];
+            if (!state["all_array_sizes"].empty())
+            {
+                state["all_array_sizes"] += ";" + current_array_sizes;
+            }
+            else
+            {
+                state["all_array_sizes"] = current_array_sizes;
+            }
+
+            // Clear the temporary states for the next member
+            state["type"].clear();
+            state["current_struct_member_name"].clear();
+            state["current_array_sizes"].clear();
+        }
+    }
+
+};
+
+#define load_type_action(Rule, id) \
+    template<> \
+    struct action<Rule> \
+    { \
+        template<typename Input> \
+        static void apply( \
+            const Input& in, \
+            Context* /*ctx*/, \
+            std::map<std::string, std::string>& state, \
+            std::vector<traits<DynamicData>::ref_type>& /*operands*/) \
+        { \
+            std::cout << "load_type_action: " << typeid(Rule).name() << " " \
+                      << in.string() << std::endl; \
```

Comment:
Please remove all debug prints, or make them conditionally compiled.
Perhaps you can convert them into EPROSIMA_LOG_INFO.